packages/beacon-node/src/util/eventLoop.ts

```diff
@@ -0,0 +1,22 @@
+import {sleep} from "@lodestar/utils";
+
+/**
+ * Schedules in 1ms a Promise to be resolved during the `timers` phase.
+ * Awaiting this Promise will force the whole event queue to be executed.
+ *
+ * Caution: as the execution of the event queue might lead to new enqueuing, this might take significant time.
+ */
+export function nextEventLoop(): Promise<void> {
+  // `setTimeout` delay is at least 1ms
+  // Say https://nodejs.org/api/timers.html#settimeoutcallback-delay-args
+  return sleep(1);
```

Comment:
Just a note for the future.
This was changed from 0 to 1 because 1 is actually the minimum that's allowed. Passing 0 here will be treated the same as passing 1.

Reply:
This seems nodejs/libuv specific behavior, might be a safer bet to just pass 0 and let the runtime specific implementation handle it. I would assume all of them handle `setTimeout 0 ` as execute in next event loop tick.